src/main/java/exercise/kadai10th/service/AirportServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package exercise.kadai10th.service;
+
+import exercise.kadai10th.entity.AirportEntity;
+import exercise.kadai10th.exceptionhandler.NoResourceException;
+import exercise.kadai10th.exceptionhandler.SameAsCurrentException;
+import exercise.kadai10th.mapper.AirportMapper;
+import exercise.kadai10th.mapper.PrefectureMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class AirportServiceImpl implements AirportService {
+
+    @Autowired
+    AirportMapper airportMapper;
+
+    @Autowired
+    PrefectureMapper prefectureMapper;
+
+    @Override
+    public AirportEntity getAirport(String airportCode) {
+        return airportMapper
+                .findByCodeFromAirports(airportCode)
+                .orElseThrow(() -> new NoResourceException(airportCode));
+    }
+
+    @Override
+    public List<AirportEntity> getAirportsByPref(String prefName) {
+        return airportMapper.findByPrefFromAirports(prefName);
+    }
+
+    @Override
+    public List<AirportEntity> getAllAirports() {
+        return airportMapper.findAllFromAirports();
+    }
+
+    @Override
+    public AirportEntity createAirport(String airportCode, String airportName, String prefCode) {
+        String prefName
+                = prefectureMapper
+                .findByCodeFromPrefs(prefCode)
+                .orElseThrow(() -> new NoResourceException(prefCode))
+                .getPrefName();
+
+        try {
+            airportMapper.insertAirport(airportCode, airportName, prefCode);
+        } catch (DuplicateKeyException e) {
+            throw new DuplicateKeyException(airportCode + " code duplicated");
+        }
+        return new AirportEntity(airportCode, airportName, prefCode, prefName);
+    }
+
+    @Override
+    public void updateAirport(String airportCode, String airportName, String prefCode) {
+        Optional<AirportEntity> optionalAirportEntity = airportMapper.findByCodeFromAirports(airportCode);
+
+        if (optionalAirportEntity.isEmpty() || prefectureMapper.findByCodeFromPrefs(prefCode).isEmpty()) {
+            throw new NoResourceException(prefCode);
```

Comment:
```suggestion
            throw new NoResourceException("airport does not exist. prefCode: " + prefCode);       
```
とかでもいいかなと思いました。
また、該当する都道府県がない場合の分岐を分けて実装してみるほうがいいかなと思います。

```java
if (airportEntity.isEmpty() {
  // 空港が存在しない事がわかる例外
}

Prefecture prefectureEntity = prefectureMapper.findByCodeFromPrefs(prefCode);
if (prefectureEntity.isEmpty()) {
  // 都道府県が存在しない事がわかる例外
}
```
throwする例外はNoResourceExceptionでよいかとは思います。
ただ、空港が存在するのに該当する都道府県が存在しないというのはデータ不整合なのでシステムエラーにすべきかもと少し思いました。

Reply:
当該箇所の処理を見直しました。

空港コード分の例外はorElseThrowメソッド中で呼び出し、都道府県コード分は別途呼び出しとしています。
空港が存在するのに該当する都道府県が存在しないケースについては、

 - ユーザーエラー：ユーザーが誤った都道府県コードを入力 ⇒ レスポンスステータスコード404
 - システムエラー：アプリ起動時に読み込んだsqlファイルの不備 ⇒ レスポンスステータスコード500
 
があると考えます。継続検討とします。

![image](https://user-images.githubusercontent.com/113277395/229475324-86b7bb3f-d26c-4471-8314-df60aa71e33b.png)
![image](https://user-images.githubusercontent.com/113277395/229474587-540dff7d-9487-420f-b7e7-075acdd279c3.png)
![image](https://user-images.githubusercontent.com/113277395/229478331-e73f797b-fba1-4850-8ab5-4078a2941e76.png)

7fbf686fa12fb90769d942df3d3fe5bdb0e477d0

Reply:
承知しました。

Reply:
異論ないです👌